modules/explorer-core/src/main/scala/org/ergoplatform/explorer/protocol/dex.scala

```diff
@@ -2,46 +2,70 @@ package org.ergoplatform.explorer.protocol
-    */
-  val sellContractTemplate: HexString = HexString(
-    refineMV[HexStringSpec](
-      "eb027300d1eded91b1a57301e6c6b2a5730200040ed801d60193e4c6b2a5730300040ec5a7eded92c1b2a57304007305720193c2b2a5730600d07307"
-    )
-  )
+  def sellContractInstance(tokensPrice: Long): ErgoTree = {
+    import org.ergoplatform.sigma.verified.VerifiedTypeConverters._
+    // since we're only using compiled contracts for constant(parameters) extraction PK value does not matter
+    val anyPk = sigmastate.eval.SigmaDsl.SigmaProp(ProveDlog(constants.group.generator))
+    AssetsAtomicExchangeCompilation.sellerContractInstance(tokensPrice, anyPk).ergoTree
+  }
 
-  // assuming seller contract from http://github.com/ScorexFoundation/sigmastate-interpreter/blob/42e55cbfd093252b8005e4607970764dd6610cbe/contract-verification/src/main/scala/sigmastate/verification/contract/AssetsAtomicExchange.scala#L33-L45
-  val BuyContractTokenIdIndexInConstants     = 6
-  val BuyContractTokenAmountIndexInConstants = 8
+  val sellContractTemplate: HexString = {
+    // parameter values does not matter, we're extracting ErgoTree template (with placeholders in places of values)
+    val contractErgoTree = sellContractInstance(0L)
+    HexString.fromString[Try](Base16.encode(ergoTreeTemplateBytes(contractErgoTree))).get
+  }
 
-  /** template for the buyer contract as of http://github.com/ScorexFoundation/sigmastate-interpreter/blob/42e55cbfd093252b8005e4607970764dd6610cbe/contract-verification/src/main/scala/sigmastate/verification/contract/AssetsAtomicExchange.scala#L12-L32
-    */
-  val buyContractTemplate: HexString = HexString(
-    refineMV[HexStringSpec](
-      "eb027300d1eded91b1a57301e6c6b2a5730200040ed803d601e4c6b2a5730300020c4d0ed602eded91b172017304938cb27201730500017306928cb27201730700027308d60393e4c6b2a5730900040ec5a7eded720293c2b2a5730a00d0730b7203"
+  def buyContractInstance(tokenId: TokenId, tokenAmount: Long): ErgoTree = {
+    import org.ergoplatform.sigma.verified.VerifiedTypeConverters._
+    // since we're only using compiled contracts for constant(parameters) extraction PK value does not matter
+    val anyPk = sigmastate.eval.SigmaDsl.SigmaProp(ProveDlog(constants.group.generator))
+    AssetsAtomicExchangeCompilation
+      .buyerContractInstance(Base16.decode(tokenId.value).get.toColl, tokenAmount, anyPk)
+      .ergoTree
+  }
+
+  val buyContractTemplate: HexString = {
+    val anyToken = TokenId(
+      "21f84cf457802e66fb5930fb5d45fbe955933dc16a72089bf8980797f24e2fa1"
     )
-  )
+    // parameter values does not matter, we're extracting ErgoTree template (with placeholders in places of values)
+    val contractErgoTree = buyContractInstance(anyToken, tokenAmount = 0L)
+    HexString.fromString[Try](Base16.encode(ergoTreeTemplateBytes(contractErgoTree))).get
```

Comment:
Same here

Reply:
Same reasons that I've described for `sellContractTemplate` apply here as well.

Reply:
Done.